02-peer-protocol.md

```diff
@@ -239,6 +239,7 @@ The receiving node MAY fail the channel if:
   - it considers `channel_reserve_satoshis` too large.
   - it considers `max_accepted_htlcs` too small.
   - it considers `dust_limit_satoshis` too small and plans to rely on the sending node publishing its commitment transaction in the event of a data loss (see [message-retransmission](02-peer-protocol.md#message-retransmission)).
```

Comment:
Should we drop this? I believe c-lightning and lnd both don't implement this at all today, and I'm not sure if we need to?

Reply:
I think it's still useful, and I kept a lower bound on eclair, because you may want to avoid letting your peer create a commit tx that you think will not easily propagate through the bitcoin network because it doesn't satisfy default policy settings.

If you think you'll rely on their commit tx with `option_dataloss_protect`, you want to be sure their commit tx can get to miners' mempools, don't you?

Reply:
> If you think you'll rely on their commit tx with option_dataloss_protect, you want to be sure their commit tx can get to miners' mempools, don't you?

Yea, that's a good point. Should we clarify that a suggested lower-bound is 330 (which IIRC is the exact current network dust threshold)?

Reply:
Done in https://github.com/lightningnetwork/lightning-rfc/pull/894/commits/f16ea995173c432400702c502633d906827267f1

Reply:
Update: I instead added a list of the network dust threshold in https://github.com/lightningnetwork/lightning-rfc/pull/894/commits/40cfede1b7f325949f8cb565cb724fdd71cf7cfc which implementers can refer to when deciding what lower bound they want to enforce.

Reply:
I don't think that's really sufficient - there needs to be guidance in what implementations should be allowed to set the dust limit to, as otherwise we end up back where we are today where some nodes arbitrarily force-closed based on dust limit and there isn't a consensus. I'd strongly prefer if we either agree to set it to the max value for a standard output, or we set it to 330 and then fix the close negotiation logic via #896.

Reply:
Right, I agree with that. Both options have pros and cons:

- if we all decide to enforce a minimum `dust_limit_satoshis = 546 sat`, we don't have issues at closing and #896 isn't even needed
- otherwise if we enforce a minimum `dust_limit_satoshis = 546 sat`, this allows smaller outputs on the commit tx but we need to do something at closing time (#896 or some other solution).

Note that an alternative to #896 which would also fix the issue would be to disallow non-segwit scripts in `shutdown`. Now that segwit is widely supported, this would fix a few headaches.

I'd like feedback from other implementers on this choice, @rustyrussell @Roasbeef what do you think?

Reply:
I opened an issue to discuss the `dust_limit_satoshis` _lower_ bounds in https://github.com/lightningnetwork/lightning-rfc/issues/905

I'd like to treat this in a separate PR than this one, which fixes a different issue that we should have fixed a long time ago (adding a _higher_ bound on our peer's `dust_limit_satoshis`).